### PR TITLE
Support extension of JSON mimetype

### LINKED ISSUE #140: Response content type json parser
Currently the deserializer only supports a list of content type values like "application/json" and "text/json".  Based on HTTP spec, "application/<randomtext>+json" is also json. So we should have a type parser to accept this syntax.

### PR DESCRIPTION
**Pull request by lmazuel**
Fix #140

### CODE REVIEW

File: msrest/pipeline/universal.py [hunk 1]
```diff
@@ -129,10 +130,9 @@ def on_response(self, request, response, **kwargs):
 
 class RawDeserializer(SansIOHTTPPolicy):
 
-    JSON_MIMETYPES = [
-        'application/json',
-        'text/json' # Because we're open minded people...
-    ]
+    # Accept "text" because we're open minded people...
+    JSON_REGEXP = re.compile(r'(application|text)/([a-z+.]+\+)?json')
```
Comment (johanste) [inline msrest/pipeline/universal.py]: This still won't match content types with parameters: 

Example: `Content-type: application/json; charset=utf-8`

Not blocking the PR for this. Just letting you know...

Comment (johanste) [reply msrest/pipeline/universal.py]: Unless the parameters have already been stripped off (which it looks like they may have been at this point)

File: tests/test_universal_pipeline.py [hunk 2]
```diff
@@ -151,6 +151,12 @@ def body(self):
...
     assert result["success"] is True
 
+    # Simple JSON with complex content_type
+    response = build_response(b'{"success": true}', content_type="application/vnd.microsoft.appconfig.kv+json")
+    raw_deserializer.on_response(None, response, stream=False)
+    result = response.context["deserialized_data"]
+    assert result["success"] is True
+
     # JSON with UTF-8 BOM
     response = build_response(b'\xef\xbb\xbf{"success": true}', content_type="application/json; charset=utf-8")
```
Comment (johanste) [inline tests/test_universal_pipeline.py]: Isn't adding a BOM in violation of https://tools.ietf.org/html/rfc7158#section-8.1?

File: msrest/pipeline/universal.py (hunk 1 above)
Comment (lmazuel) [reply msrest/pipeline/universal.py]: Yes, they have at this point (see a little later in the file).

File: tests/test_universal_pipeline.py (hunk 2 above)
Comment (lmazuel) [reply tests/test_universal_pipeline.py]: https://github.com/Azure/msrest-for-python/issues/144

**Pull request merged**